src/sql/src/plan/statement/ddl.rs

```diff
@@ -911,7 +916,55 @@ pub fn plan_create_source(
             if key_envelope == KeyEnvelope::None {
                 key_envelope = get_unnamed_key_envelope(key_encoding)?;
             }
-            UnplannedSourceEnvelope::Upsert(UpsertStyle::Default(key_envelope))
+
+            let order_by = if !order_by.is_empty() {
+                scx.require_unsafe_mode("ENVELOPE UPSERT ORDER BY")?;
+
+                let mut offset_included = false;
+
+                let metadata_indices = order_by
+                    .iter()
+                    .map(|order_col| {
+                        let metadata_type = metadata_columns
+                            .iter()
+                            .position(|(col_name, _)| *col_name == order_col.as_str())
+                            .map(|pos| {
+                                let included_col = metadata_column_types
+                                    .get(pos)
+                                    .expect("metadata index calculation error");
```

Comment:
is this guaranteed to be the same as `metadata_columns`? I assume yes

Reply:
That's what I parsed from https://github.com/MaterializeInc/materialize/blob/main/src/storage-client/src/types/sources.rs#L1434-L1494, unless I am reading it wrong.